File: tests/fixtures/sample.kt
package com.example

class Service(val name: String) {
    private val secret: String = ""
    var value: Int = 0

    fun getName(): String {
        return name
    }

    private fun internal() {
        // hidden
    }

    companion object {
        fun create(name: String): Service {
            return Service(name)
        }
    }
}

object Singleton {
    fun doWork() {
        println("working")
    }
}

enum class Color {
    RED,
    GREEN,
    BLUE;

    fun label(): String {
        return name.lowercase()
    }
}

typealias Callback = (String) -> Unit

fun greet(name: String): String {
    return "Hello, $name!"
}

val MAX_SIZE = 1024

File: tests/fixtures/sample.swift
import Foundation

class Service {
    var name: String
    private var secret: String

    init(name: String) {
        self.name = name
        self.secret = ""
    }

    func getName() -> String {
        return name
    }

    private func internal() {
        // hidden
    }
}

protocol Reader {
    func read(buffer: Data) -> Int
    func close()
}

struct Config {
    var name: String
    var value: Int
}

enum Color {
    case red
    case green
    case blue

    func label() -> String {
        switch self {
        case .red: return "Red"
        case .green: return "Green"
        case .blue: return "Blue"
        }
    }
}

func greet(name: String) -> String {
    return "Hello, \(name)!"
}

typealias Callback = (String) -> Void
